cmd/snapshot-create.go

```diff
@@ -0,0 +1,161 @@
+		// Initialize logger (in a real scenario, you'd likely pass in a config)
+		l, _ := logger.NewLogger(&logger.LoggerConfig{Debug: true})
+
+		// Retrieve database connection details from global flags
+		dbHost, _ := cmd.Flags().GetString("database.host")
+		dbName, _ := cmd.Flags().GetString("database.db_name")
+		dbUser, _ := cmd.Flags().GetString("database.user")
+		dbPassword, _ := cmd.Flags().GetString("database.password")
+		dbPort, _ := cmd.Flags().GetInt("database.port")
+		schemaName, _ := cmd.Flags().GetString("database.schema_name")
+
+		// Retrieve chain and fidelity
+		chain, _ := cmd.Flags().GetString("chain")
+
+		// Validate fidelity
+		if snapshotFidelity != "archive" {
+			l.Sugar().Warnw("Unsupported fidelity specified; falling back to 'archive'.",
+				"requested", snapshotFidelity,
+				"used", "archive",
+			)
+			snapshotFidelity = "archive"
+		}
+
+		// Log database connection details at info level
+		l.Sugar().Infow("Database connection details",
+			"host", dbHost,
+			"name", dbName,
+			"user", dbUser,
+			"port", dbPort,
+			"schema", schemaName,
+		)
+
+		// If no specific file name is provided, generate one
+		if snapshotCreateToPath == "" {
+			fileNameSuffix := time.Now().UTC().Format("2006-01-02-15-04-05")
+			snapshotCreateToPath = fmt.Sprintf("sidecar-snapshot-%s-%s-%s.dump", chain, snapshotFidelity, fileNameSuffix)
+		}
+
+		// Log the snapshot file path
+		l.Sugar().Infow("Snapshot will be created at this file", "file", snapshotCreateToPath)
+
+		// Set the password in an environment variable for pg_dump
+		if err := os.Setenv("PGPASSWORD", dbPassword); err != nil {
+			l.Sugar().Fatalw("Failed to set environment variable", "error", err)
+			return err
+		}
+
+		// Construct the pg_dump command with a context for timeout
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Minute)
+		defer cancel()
+
+		pgDumpCmd := exec.CommandContext(ctx,
+			"pg_dump",
+			"--host", dbHost,
+			"--port", fmt.Sprintf("%d", dbPort),
+			"--username", dbUser,
+			"--format", "custom",
+			"--file", snapshotCreateToPath,
+			"--schema", schemaName,
+			dbName,
+		)
+
+		// Direct command output to stdout/stderr for visibility
+		pgDumpCmd.Stdout = os.Stdout
+		pgDumpCmd.Stderr = os.Stderr
+
+		l.Sugar().Infow("Running pg_dump command",
+			"command", pgDumpCmd.String(),
+		)
+
+		// Run the pg_dump command
+		if err := pgDumpCmd.Run(); err != nil {
+			if ctx.Err() == context.DeadlineExceeded {
+				l.Sugar().Errorw("pg_dump command timed out",
+					"error", err,
+					"command", pgDumpCmd.String(),
+				)
+				return fmt.Errorf("pg_dump command timed out: %w", err)
+			}
+			l.Sugar().Errorw("Failed to run pg_dump",
+				"error", err,
+				"command", pgDumpCmd.String(),
+			)
+			return fmt.Errorf("failed to run pg_dump: %w", err)
+		}
+
+		l.Sugar().Infow("Successfully created snapshot", "file", snapshotCreateToPath)
+
+		// Compress the snapshot if the flag is set
+		if compressSnapshot {
```

Comment:
Rather than downloading the file to the filesystem and THEN compressing it, Id suggest looking into how to pipeline the process to minimize how much we write then read to disk. If you can take the stream from `pg_dump` and optionally pipe it directly to `tar`, thats much preferred than writing to disk first THEN compressing